File: RB101_programming_foundations/lesson_5/5_practice_problems/ex_13.rb
# Given the following data structure, return a new array containing the same sub-arrays as the original but ordered logically by only taking into consideration the odd numbers they contain.

# The sorted array should look like this:

# [[1, 8, 3], [1, 6, 7], [1, 4, 9]]

arr = [[1, 6, 7], [1, 4, 9], [1, 8, 3]]

arr.sort_by do |subarr|
  subarr.select { |num| num.odd? }
end

File: JS101_programming_foundations/lesson_5/practice_problems/ex_13.js
// Given the following data structure, sort the array so that the sub - arrays are ordered based on the sum of the odd numbers that they contain.

let arr = [[1, 6, 7], [1, 5, 3], [1, 8, 3]];

arr.sort((a, b) => {
  a = a.reduce((sum, num) => {
    if (num % 2 === 1) {
      sum += num;
    }
    return sum;
  }, 0);

  b = b.reduce((sum, num) => {
    if (num % 2 === 1) {
      sum += num;
    }
    return sum;
  }, 0);

  return a - b;
})

console.log(arr);
